pkg/services/rendering/auth.go

```diff
@@ -18,15 +24,52 @@ type RenderUser struct {
 	OrgRole string
 }
 
+type renderJWT struct {
+	RenderUser *RenderUser
+	jwt.RegisteredClaims
+}
+
 func (rs *RenderingService) GetRenderUser(ctx context.Context, key string) (*RenderUser, bool) {
-	val, err := rs.RemoteCacheService.Get(ctx, fmt.Sprintf(renderKeyPrefix, key))
+	var (
+		from   string
+		exists bool
+		err    error
+		start  = time.Now()
+	)
+
+	defer func() {
+		success := strconv.FormatBool(err == nil && exists)
+		metrics.MRenderingUserLookupSummary.WithLabelValues(success, from).Observe(float64(time.Since(start)))
+	}()
+
+	if looksLikeJWT(key) && rs.features.IsEnabled(featuremgmt.FlagRenderingOverJWT) {
+		from = "jwt"
+		claims := new(renderJWT)
+		var tkn *jwt.Token
+		tkn, err = jwt.ParseWithClaims(key, claims, func(_ *jwt.Token) (interface{}, error) {
+			return []byte(rs.Cfg.RendererAuthToken), nil
+		})
+
+		if err != nil || !tkn.Valid {
+			rs.log.Error("Could not get render user from JWT", "err", err)
+			return nil, exists
+		}
+
+		exists = true
+		return claims.RenderUser, exists
```

Comment:
I'm confused, I don't see us validating the jwt signature anywhere here?

Reply:
I thought (and tested) that the `Parse` method performs certain validations.

But, it's true that re-reading its docs ([here](https://pkg.go.dev/github.com/golang-jwt/jwt/v4#Parse)), they strongly encourage to validate the "alg", as well.

Is that what you mean? Thanks!

Reply:
I was confused, I didn't see the keyFunc implementation providing the configured key.

We should definitely set the WithValidMethods option in the call to ParseWithClaims or check the alg inside keyFunc.